apps/veil/src/entities/leaderboard/api/use-lp-leaderboard.ts

```diff
@@ -0,0 +1,73 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { Position } from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+import { apiPostFetch } from '@/shared/utils/api-fetch';
+import {
+  LpLeaderboardRequest,
+  LpLeaderboardApiResponse,
+  LpLeaderboardSortKey,
+  LpLeaderboardSortDirection,
+  LqtLp,
+  LpLeaderboard,
+  LpLeaderboardResponse,
+} from '@/entities/leaderboard/api/utils';
+import { penumbra } from '@/shared/const/penumbra';
+import { DexService } from '@penumbra-zone/protobuf';
+import { bech32mPositionId } from '@penumbra-zone/bech32m/plpid';
+
+export const BASE_LIMIT = 10;
+export const BASE_PAGE = 1;
+
+// get the position state for each lp reward
+async function enrichLpLeaderboards(data: LqtLp[]): Promise<LpLeaderboard[]> {
+  if (data.length === 0) {
+    return [];
+  }
+
+  const positionIds = data.map(lp => lp.positionId);
+  const positionsRes = await Array.fromAsync(
+    penumbra.service(DexService).liquidityPositionsById({ positionId: positionIds }),
+  );
+  const positions = positionsRes.map(r => r.data).filter(Boolean) as Position[];
+
+  return data.map((lp, index) => ({
+    ...lp,
+    positionIdString: bech32mPositionId(lp.positionId),
+    position: positions[index] as unknown as Position,
+  }));
+}
+
+export const useLpLeaderboard = ({
+  epoch,
+  page,
+  limit,
+  sortKey,
+  sortDirection,
+  isActive,
+}: {
+  epoch: number | undefined;
+  page: number;
+  limit: number;
+  sortKey?: LpLeaderboardSortKey | '';
+  sortDirection?: LpLeaderboardSortDirection;
+  isActive: boolean;
+}): UseQueryResult<LpLeaderboardResponse> => {
+  const query = useQuery({
+    queryKey: ['lp-leaderboard', epoch, page, limit, sortKey, sortDirection],
+    staleTime: Infinity,
+    queryFn: async () => {
+      return apiPostFetch<LpLeaderboardApiResponse>('/api/tournament/lp-leaderboard', {
+        epoch,
+        page,
+        limit,
+        sortKey,
+        sortDirection,
+      } as LpLeaderboardRequest).then(async resp => ({
+        ...resp,
+        data: await enrichLpLeaderboards(resp.data),
+      }));
+    },
+    enabled: typeof epoch === 'number' && isActive,
```

Comment:
suggestion: refetch the leaderboard on a new block if the epoch is not over. Disable refetch if epoch has ended